src/models/Step.js

```diff
@@ -22,7 +22,7 @@ StepSchema.virtual('subProgram', {
 
 // eslint-disable-next-line consistent-return
 function setAreActivitiesValid() {
-  if (this.activities && this.activities.length === 0) return true;
+  if (this.type === ON_SITE && this.activities && this.activities.length === 0) return true;
```

Comment:
idem ici tu peux faire 
```
const hasActivities = this.activities && this.activities.length !== 0
if (this.type === ON_SITE && !hasActivities) return true;
if (this.type === E_LEARNING && !hasActivities) return true;
```

Reply:
`if (this.type === E_LEARNING && !hasActivities) return true;` est à enlever non? car on ne veut pas renvoyer `true` pour un elearning step sans activités

Reply:
return false pardon